tornado/simple_httpclient.py

```diff
@@ -491,9 +491,17 @@ def headers_received(self, first_line, headers):
             self.request.header_callback('\r\n')
 
     def _should_follow_redirect(self):
-        return (self.request.follow_redirects and
+        if not (self.request.follow_redirects and
                 self.request.max_redirects > 0 and
-                self.code in (301, 302, 303, 307, 308))
+                self.code in (301, 302, 303, 307, 308)):
+            return False
+        # http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.3.4
+        # According to the spec, the temporary URI may not be specified when the
```

Comment:
I think you're misreading the spec. The Location header SHOULD be given no matter what the request method was. The only difference with HEAD is whether there should be a body for clients that don't follow redirects automatically (there's nothing special about HEAD with 302 in this respect; that's how HEAD always works).

Also, RFC2616 is obsolete. The current definition of the 302 response code is https://tools.ietf.org/html/rfc7231#section-6.4.3.